File: pset2/Problem Set 2/Readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h>
#include <math.h>
int main(void)
{
    int letters = 0, i = 0, sentences = 0, m = 0, n = 0, words = 0, index;
    float L, S;
    string para = get_string("Text : ");
    n = strlen(para);
    for (i = 0 ; n > i ; i++)
    {
        char p = para[i];//Checking for letters
        if (isalpha(p))
        {
            letters += 1;
        }
        if (isspace(p) && !isspace(para[i - 1]))//check for words
        {
            words++;
        }
        if (para[i] == '.' || para[i] == '?' || para[i] == '!')//check for sentences
        {
            sentences++;
        }
    }

    words = words + 1;
    S = (sentences / (float) words) * 100;
    L = (letters / (float) words) * 100;
    index = round(0.0588 * L - 0.296 * S - 15.8); // comparing and applying the formula 

    if (index < 1) //finding the grade
    
    {
        printf("Before Grade 1\n");
        
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
        
    }
    else
    {
        printf("Grade %d\n", index);//the answer
    }

}

File: pset6/readability/readability.py
from cs50 import get_string
para = get_string("Text: ")
z = 0
s = 0
l = 0
for ch in para:
    if (ch.isalpha()):  # for letters in ch
        l = l + 1
    if (ch == ' '):  # for words in ch
        s = s + 1
    if (ch == '.' or ch == '?' or ch == '!'):  # for sentences in ch
        z = z + 1
s = s + 1
z = (z / float.(s)) * 100
l = (l / float(s)) * 100
index = round(0.0588 * l - 0.296 * z - 15.8)  # applying formula
if (index > 16):
    print("Grade 16+")
if (index < 1):
    print("Before Grade 1")
else:
    print(f"Grade {index}")